src/components/spring.js

```diff
@@ -0,0 +1,103 @@
+var CANNON = require('cannon');
+
+module.exports = AFRAME.registerComponent("spring", {
+
+  multiple: true,
+
+  schema: {
+    // Target (other) body for the constraint.
+    target: {type: 'selector'},
+
+    // Lenght of the spring, when no force acts upon it
+    restLength: {default: 1, min: 0},
+
+    // how much will the spring suppress the force
+    stiffness: {default: 100, min: 0, max: 0},
+
+    // the stretch factor of the spring
+    damping: {default: 1, min: 0, max: 1},
+
+    // offsets
+    localAnchorA: {type: 'vec3', default: {x: 0, y: 0, z: 0}},
+    localAnchorB: {type: 'vec3', default: {x: 0, y: 0, z: 0}},
+  },
+
+  init: function() {
+    this.system = this.el.sceneEl.systems.physics
+    this.world = this.system.driver.world
```

Comment:
Is it possible to implement this without touching `system.driver` or `system.driver.world`? If so, it makes it much easier to adapt this to work with Web Workers or future physics engines... in theory you should be able to do `this.system.addComponent(this)` in init, and `this.system.removeComponent(this)` in remove, and then define `beforeStep` and `afterStep` methods on the component where we update the spring. See `system.js`.

Reply:
Thanks for the hint, not only did I reimplement it using the system callbacks, but also it seems to be working 😄